--- a/ipu6-drivers/drivers/media/pci/intel/ipu6/ipu6-isys-csi2.c
+++ b/ipu6-drivers/drivers/media/pci/intel/ipu6/ipu6-isys-csi2.c
@@ -42,6 +42,8 @@ static const u32 csi2_supported_codes[] = {
 	MEDIA_BUS_FMT_SGBRG8_1X8,
 	MEDIA_BUS_FMT_SGRBG8_1X8,
 	MEDIA_BUS_FMT_SRGGB8_1X8,
+	MEDIA_BUS_FMT_VYUY8_1X16, /* D4XX specific */
+	MEDIA_BUS_FMT_Y8_1X8, /* D4XX specific */
 	MEDIA_BUS_FMT_META_8,
 	MEDIA_BUS_FMT_META_10,
 	MEDIA_BUS_FMT_META_12,
--- a/ipu6-drivers/drivers/media/pci/intel/ipu6/ipu6-isys-csi2.h
+++ b/ipu6-drivers/drivers/media/pci/intel/ipu6/ipu6-isys-csi2.h
@@ -21,7 +21,7 @@ struct ipu6_isys_stream;
 #define INVALID_VC_ID		-1
 #define NR_OF_CSI2_SINK_PADS	1
 #define CSI2_PAD_SINK		0
-#define NR_OF_CSI2_SRC_PADS	8
+#define NR_OF_CSI2_SRC_PADS	8 /* D4XX specific; modify this number to support more VC */
 #define CSI2_PAD_SRC		1
 #define NR_OF_CSI2_PADS		(NR_OF_CSI2_SINK_PADS + NR_OF_CSI2_SRC_PADS)

--- a/ipu6-drivers/drivers/media/pci/intel/ipu6/ipu6-isys-subdev.c
+++ b/ipu6-drivers/drivers/media/pci/intel/ipu6/ipu6-isys-subdev.c
@@ -24,6 +24,7 @@ unsigned int ipu6_isys_mbus_code_to_bpp(u32 code)
 	case MEDIA_BUS_FMT_RGB565_1X16:
 	case MEDIA_BUS_FMT_UYVY8_1X16:
 	case MEDIA_BUS_FMT_YUYV8_1X16:
+	case MEDIA_BUS_FMT_VYUY8_1X16:
 	case MEDIA_BUS_FMT_META_16:
 		return 16;
 	case MEDIA_BUS_FMT_SBGGR12_1X12:
@@ -43,6 +44,7 @@ unsigned int ipu6_isys_mbus_code_to_bpp(u32 code)
 	case MEDIA_BUS_FMT_SGBRG8_1X8:
 	case MEDIA_BUS_FMT_SGRBG8_1X8:
 	case MEDIA_BUS_FMT_SRGGB8_1X8:
+	case MEDIA_BUS_FMT_Y8_1X8:
 	case MEDIA_BUS_FMT_META_8:
 		return 8;
 	default:
@@ -60,6 +62,7 @@ unsigned int ipu6_isys_mbus_code_to_mipi(u32 code)
 		return MIPI_CSI2_DT_RGB888;
 	case MEDIA_BUS_FMT_UYVY8_1X16:
 	case MEDIA_BUS_FMT_YUYV8_1X16:
+	case MEDIA_BUS_FMT_VYUY8_1X16:
 		return MIPI_CSI2_DT_YUV422_8B;
 	case MEDIA_BUS_FMT_SBGGR16_1X16:
 	case MEDIA_BUS_FMT_SGBRG16_1X16:
@@ -82,6 +85,7 @@ unsigned int ipu6_isys_mbus_code_to_mipi(u32 code)
 	case MEDIA_BUS_FMT_SGBRG8_1X8:
 	case MEDIA_BUS_FMT_SGRBG8_1X8:
 	case MEDIA_BUS_FMT_SRGGB8_1X8:
+	case MEDIA_BUS_FMT_Y8_1X8:
 		return MIPI_CSI2_DT_RAW8;
 	default:
 		/* return unavailable MIPI data type - 0x3f */
--- a/ipu6-drivers/drivers/media/pci/intel/ipu6/ipu6-isys-video.c
+++ b/ipu6-drivers/drivers/media/pci/intel/ipu6/ipu6-isys-video.c
@@ -88,6 +88,15 @@ const struct ipu6_isys_pixelformat ipu6_isys_pfmts[] = {
 	  IPU6_FW_ISYS_FRAME_FORMAT_RGB565 },
 	{ V4L2_PIX_FMT_BGR24, 24, 24, MEDIA_BUS_FMT_RGB888_1X24,
 	  IPU6_FW_ISYS_FRAME_FORMAT_RGBA888 },
+	/* D4XX specific */
+	{ V4L2_PIX_FMT_Z16, 16, 16, MEDIA_BUS_FMT_UYVY8_1X16,
+	  IPU6_FW_ISYS_FRAME_FORMAT_UYVY },
+	{ V4L2_PIX_FMT_Y8I, 16, 16, MEDIA_BUS_FMT_VYUY8_1X16,
+	  IPU6_FW_ISYS_FRAME_FORMAT_UYVY },
+	{ V4L2_PIX_FMT_Y12I, 32, 24, MEDIA_BUS_FMT_RGB888_1X24, /* TODO: To be verified */
+	  IPU6_FW_ISYS_FRAME_FORMAT_RGBA888 },
+	{ V4L2_PIX_FMT_GREY, 8, 8, MEDIA_BUS_FMT_Y8_1X8,
+	  IPU6_FW_ISYS_FRAME_FORMAT_RAW8 },
 	{ V4L2_META_FMT_GENERIC_8, 8, 8, MEDIA_BUS_FMT_META_8,
 	  IPU6_FW_ISYS_FRAME_FORMAT_RAW8, true },
 	{ V4L2_META_FMT_GENERIC_CSI2_10, 10, 10, MEDIA_BUS_FMT_META_10,
--- a/ipu6-drivers/drivers/media/pci/intel/ipu6/ipu6-isys.h
+++ b/ipu6-drivers/drivers/media/pci/intel/ipu6/ipu6-isys.h
@@ -46,4 +46,4 @@ struct ipu6_isys_pdata;
 #define IPU6_ISYS_MIN_WIDTH		2U
-#define IPU6_ISYS_MIN_HEIGHT		2U
+#define IPU6_ISYS_MIN_HEIGHT		1U /* D4XX specific */
 #define IPU6_ISYS_MAX_WIDTH		4672U
 #define IPU6_ISYS_MAX_HEIGHT		3416U
