diff --git a/drivers/media/i2c/max9x/serdes.c b/drivers/media/i2c/max9x/serdes.c
index b6e38639f..ba888a07a 100644
--- a/drivers/media/i2c/max9x/serdes.c
+++ b/drivers/media/i2c/max9x/serdes.c
@@ -1858,13 +1858,16 @@ static int max9x_registered(struct v4l2_subdev *sd)
 						.dev_id = "",
 						.table = {
 							GPIO_LOOKUP("", 0, "reset",
-								    GPIO_ACTIVE_HIGH),
+								    GPIO_ACTIVE_LOW),
+							GPIO_LOOKUP("", 7, "fsin",
+								    GPIO_ACTIVE_LOW),
 							{}
 						},
 					};
 
 					sensor_gpios.dev_id = dev_id;
 					sensor_gpios.table[0].key = common->gpio_chip.label;
+					sensor_gpios.table[1].key = common->gpio_chip.label;
 
 					gpiod_add_lookup_table(&sensor_gpios);
 
